packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+                     |                 |            |     |              |
+                     +--------^--------+            |     +--------------+
+                              |                     |
+                              |                     |
+                        +-----v----+                |     +----------+
+                        |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  API Request   |     |          |
+                        |          <-----------------------          |
+                        |          |  API Response  |     |          |
+         +---------+    |          ----------------------->          |
+         |         |    |          |                |     |          |
+         |  STORE  <--->|  WALLET  |  Events        |     |   APP    |
+         |         |    |          ----------------------->          |
+         +---------+    |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  pushMessage   |     |          |
+                        |          <---------------------------+     |
+                        |          |                |     |    |     |
+                        |          |                |     |    |     |
+                        +----------+                |     +----|-----+
+                                                    |          |
+                                                    | +--------|---------+
+                                                    | |  COUNTERPARTY    |
+                                                    | |  WALLETS         |
+                                                      +------------------+
+```
+
+# The responsibilities of a state channel wallet
+
+As in the [Hippocratic oath](https://en.wikipedia.org/wiki/Hippocratic_Oath): _primum non nocere,_ (first, do no harm).
+
+1. Do not lose secrets (private keys)
+2. Do not leak secrets (private keys)
+3. Store relevant states and signatures (Don’t lose any unless they are screened off by kept ones, allow states to be queried)
+4. Do not sign states or blockchain transactions unless the App grants permission
+
+Next, do some active things to protect The App’s interests.
+
+5. With implicit permission, allow an exit on chain without losing funds
+6. Detect and store blockchain challenges (for querying)
+7. Detect and store changes in channel funding (for querying)
+
+Next, do some active things to allow The App to express itself
+
+8. With explicit permission, sign updates in running application channels
+9. With implicit permission, fund channels on chain
+
+Next, allow for advanced funding relationships
+
+10. With implicit permission, sign “safe” updates in ledger channels (meaning no funds are lost)
```

Comment:
A wallet should also sign updates in eg. guarantor & joint channels.

```suggestion
10. With implicit permission, sign “safe” updates in "auxilliary" funding channels, such as a ledger channel

 "Safe" here means that no funds are lost
```

Reply:
I think this is an example of where I interpret the terminology slightly differently to others. I tend to think of anything that is not an application channel as a ledger channel.

From the nitro paper: 
> A ledger channel is a channel whose sole purpose is to provide funding to
other channels. 

On the other hand, I think many other people do not interpret it this way. Also your recommendation works under both interpretations. So I went with it!

Reply:
Is a guarantor channel a ledger channel?